src/Sulu/Component/Content/Document/Subscriber/WebspaceSubscriber.php

```diff
@@ -49,13 +62,39 @@ public function supports($document)
     }
 
     /**
-     * @param HydrateEvent $event
+     * @param AbstractMappingEvent|HydrateEvent $event
+     * @throws \Sulu\Component\DocumentManager\Exception\DocumentManagerException
      */
     public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
-        $webspace = $this->inspector->getWebspace($document);
-        $event->getAccessor()->set('webspaceName', $webspace);
+        $webspaceName = $this->inspector->getWebspace($document);
+        $event->getAccessor()->set('webspaceName', $webspaceName);
+
+        if ($document instanceof StructureBehavior && $document->getStructureType() === null) {
+            $this->setDefaultStructureType($event, $document, $webspaceName);
+        }
```

Comment:
I guess I create an issue for this, and just correct the tests for now. I'll see then if this can be solved easier after merging #1272. Btw, there are still your own comments you should probably fix :wink: